### PR TITLE
[backend/frontend] Search API for Contract, Attack Pattern, Kill Chain Phases and Dynamic Asset Group

### PR DESCRIPTION
**Pull request by RomuDeuxfois**
Related issue: https://github.com/OpenBAS-Platform/openbas/issues/559
Related issue: https://github.com/OpenBAS-Platform/openbas/issues/510

Final version of a global pagination system with filtering, searching and sorting: runtime & JPA mechanism.
Add search on Attack pattern and Kill Chain Phases screen.
Add export button with new style on Attack pattern and Kill Chain Phases screen.
Add dynamic asset group feature

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/utils/pagination/SortUtilsRuntime.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import static java.util.Comparator.comparing;
+import static org.springframework.util.StringUtils.hasText;
+
+public class SortUtilsRuntime {
+
+  public static Sort toSortRuntime(@Nullable final List<SortField> sorts) {
+    List<Sort.Order> orders;
+
+    if (sorts == null || sorts.isEmpty()) {
+      orders = List.of();
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/pagination/SortUtilsRuntime.java:24]: What is the default value if sorting is not defined? By what property are the results sorted?

File: openbas-api/src/main/java/io/openbas/rest/kill_chain_phase/KillChainPhaseApi.java [hunk 2]
```diff
@@ -4,47 +4,60 @@
...
-    }
+  private KillChainPhaseRepository killChainPhaseRepository;
+
+  @Autowired
+  public void setKillChainPhaseRepository(KillChainPhaseRepository killChainPhaseRepository) {
+    this.killChainPhaseRepository = killChainPhaseRepository;
+  }
+
+  @GetMapping("/api/kill_chain_phases")
+  public Iterable<KillChainPhase> killChainPhases() {
```
[the remark this answers is not in the document]
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/rest/kill_chain_phase/KillChainPhaseApi.java:32]: We already follow the nomenclature with _ on other urls..

File: openbas-api/src/main/java/io/openbas/utils/pagination/SortUtilsRuntime.java (hunk 1 above)
Comment (RomuDeuxfois) [reply openbas-api/src/main/java/io/openbas/utils/pagination/SortUtilsRuntime.java:24]: If it's not defined, there are no sorting

Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/utils/pagination/SortUtilsRuntime.java:24]: ok, sorted as ddbb result (maybe by PK)

**Comment by savacano28** [comment]
Final version of a global pagination system with filtering, searching and sorting: runtime & JPA mechanism: Tested in contract pagination (en, fr, with sortFields[])
Add search on Attack pattern and Kill Chain Phases screen: ok
Add export button with new style on Attack pattern and Kill Chain Phases screen: ok

**Review by savacano28 — APPROVED** [review]

**Pull request merged**